app/models/authentication.rb

```diff
@@ -25,4 +25,23 @@ def set_source
                        resource.try(:source_id)
                      end
   end
+
+  def update_status
+    super
+
+    if availability_status.nil? || changed.include?(:availability_status)
+      reset_related_applications_status
+      reset_related_endpoint_status
+    end
+  end
+
+  def reset_related_applications_status
+    applications.each(&:remove_availability_status!)
```

Comment:
Shouldn't this be `resource` as well? So we could just have one method that removes the sub-resource's availability_status

Reply:
this call should be moved above the return (because it's about application_authentications relation)
and the `resource` call should be added